git-russian-roulette

```diff
@@ -23,7 +21,8 @@ git stash clear > /dev/null 2>&1
 
 # get a random directory or file from all git tracked files
 # and mark it as sentenced for dead.
-SENTENCED_FILE=$(git ls-tree -r HEAD --full-name --name-only | sort -R | head -n 1)
+ALL_FILES=( $(git ls-tree -r HEAD --full-name --name-only | tr '\n' ' ') )
```

Comment:
About the `tr`: It's not exactly valid. If a filename contains a space it will try to remove a file that does not exist. use `IFS=$'\n'` before `ALL_FILES` assignment and it's ok. So simply add IFS to use only newline characters and remove the `tr` call.

Read more about IFS: https://bash.cyberciti.biz/guide/$IFS